source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -1,79 +1,717 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using System.Reflection;
+using System.Text;
 using Calamari.Common.Features.EmbeddedResources;
 using Calamari.Common.Features.Processes;
 using Calamari.Common.Features.Scripting;
 using Calamari.Common.Features.Scripts;
+using Calamari.Common.Plumbing;
+using Calamari.Common.Plumbing.Commands;
 using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Proxies;
 using Calamari.Common.Plumbing.Variables;
+using YamlDotNet.RepresentationModel;
 
 namespace Calamari.Kubernetes
 {
     public class KubernetesContextScriptWrapper : IScriptWrapper
```

Comment:
Awesome effort wrangling this thing out. There is a lot of behaviour in here, maintaining that in two places would have been tough going!